src/audio/multiband_drc/multiband_drc.c

```diff
@@ -398,15 +398,34 @@ static int multiband_drc_init(struct processing_module *mod)
 	return ret;
 }
 
+/**
```

Comment:
The commit `MDRC: Improve Resource Cleanup in multiband_drc_free` is not useful, `cd` cannot be NULL on creation exit therefore the `if` check adds no value.

Reply:
@ShriramShastry in many cases checking for impossible conditions doesn't add security, instead it adds confusion (provided the condition is indeed provably impossible)

Reply:
Second @lyakh, its not good practice to add code for conditions that are impossible by design. it wastes space and confuses other devs who might use this as a reference when they build their code.

Reply:
Sure, then I will remove it.

Reply:
you still have the if case in the free, please remove it.